pyEpiabm/pyEpiabm/sweep/assign_household_ages_sweep.py

```diff
@@ -0,0 +1,389 @@
+                people[0].set_random_age()
+                people[1].set_random_age()
+                if ((people[1].age >= self.age_params["min_adult_age"]) and
+                        (people[0].age >= self.age_params["min_adult_age"])):
+                    break
+            while True:
+                people[2].set_random_age()
+                if ((people[2].age <= people[1].age
+                     + self.age_params["max_MF_partner_age_gap"]) and
+                        (people[2].age >= people[1].age
+                         - self.age_params["max_FM_partner_age_gap"])):
+                    break
+
+        # set ages when children are members of a household
+        else:
+            while True:
+                people[0].age = 0
+                # set ages of children, increasing the ages between them if
+                # there are multiple children in the house
+                for i in range(1, num_childs):
+                    people[i].age = (
+                        people[i-1].age + 1
+                        + np.random.poisson((self.age_params
+                                            ["mean_child_age_gap"] - 1)))
+                random_child_index = int(np.floor(
+                                         random.random() * num_childs))
+                age_group = random.choices(range(self.num_age_groups),
+                                           weights=self.age_proportions)[0]
+                age = random.randint(0, 4) + 5 * age_group
+                people[0].age = age - people[random_child_index].age
+                # increase base age of children relative to youngest child
+                for i in range(1, num_childs):
+                    people[i].age += people[0].age
+                # set max age of youngest child
+                youngest_age = 5 if \
+                    (((num_childs == 1) and
+                        (random.random()
+                         < self.age_params
+                         ["one_child_prob_youngest_child_under_five"])) or
+                     ((num_childs == 2) and
+                        (random.random()
+                            < self.age_params
+                            ["two_children_prob_youngest_under_five"]))) else \
+                    self.age_params["max_child_age"]
+                if ((people[0].age >= 0) and
+                        (people[0].age <= youngest_age) and
+                        (people[num_childs - 1].age
+                         <= self.age_params["max_child_age"])):
+                    break
+
+            # sets age difference of parents and children
+            parent_age_gap = (people[num_childs - 1].age - people[0].age
+                              - (self.age_params["max_parent_age_gap"]
+                              - self.age_params["min_parent_age_gap"]))
+            if parent_age_gap > 0:
+                parent_age_gap += \
+                    self.age_params["max_parent_age_gap"]  # pragma: no cover
```

Comment:
I am still not 100% on this - we haven't used this anywhere else. Remind me what case this covers for again (or I'll look at the wiki when it is up)

Reply:
The pragma no cover in the host progression sweep was for a `continue` statement that was actually covered, but not included in the calculations due to a error on their end, whereas I think this is a little different. I'll have another look at this case once the wiki is up to date with the parameter descriptions to see if I can work out what it is handling, and whether we can mock it.